--- ngsw-worker.js.orig	1985-10-26 13:45:00.000000000 +0530
+++ ngsw-worker.js	2019-02-26 18:55:33.399888700 +0530
@@ -2023,6 +2023,24 @@
             return __awaiter$5(this, void 0, void 0, function* () {
                 notification.close();
                 const options = {};
+
+                let url = 'https://insti.app/';
+                const type = notification.data.type
+                if (type === 'blogentry') {
+                    if (notification.data.extra.indexOf('training') !== -1)
+                        url += 'blog/ImFwaS90cmFpbmluZy1ibG9nez9mcm9tLG51bSxxdWVyeX0i';
+                    else
+                        url += 'blog/ImFwaS9wbGFjZW1lbnQtYmxvZ3s%252FZnJvbSxudW0scXVlcnl9Ig%253D%253D';
+                } else if (type === 'newsentry') {
+                    url += 'news'
+                } else if (type === 'event') {
+                    url += 'event/' + notification.data.id;
+                }
+                url += '?notif=' + notification.notification_id;
+
+                clients.openWindow(url);
+
+
                 // The filter uses `name in notification` because the properties are on the prototype so
                 // hasOwnProperty does not work here
                 NOTIFICATION_OPTION_NAMES.filter(name => name in notification)
